src/Callback.php

```diff
@@ -164,4 +176,77 @@ private function schedule_callback( $session_id ) {
 
 		return 0 !== $did_schedule;
 	}
+
+	/**
+	 * Handles notification callbacks
+	 *
+	 * @param $request
+	 *
+	 * @return \WP_REST_Response
+	 */
+	public function handle_notification( $request ) {
+		$context = array(
+			'filter'   => current_filter(),
+			'function' => __FUNCTION__,
+		);
+
+		$request_body = json_decode( $request->get_body() );
+		$response     = new \WP_REST_Response();
+
+		if ( json_last_error() !== JSON_ERROR_NONE ) {
+			Ledyer_Payments()->logger()->debug( '[NOTIFICATION]: Request body isn\'t valid JSON string.', $context );
+			$response->set_status( 400 );
+			return $response;
+		}
+
+		$ledyer_event_type = $request_body->{'eventType'};
+		$ledyer_order_id   = $request_body->{'orderId'};
```

Comment:
This works, but id prefer to decode the json body as an array and read the values from that. If we kept it as an object we should just get the values by calling `$request_body->eventType` as well, since without decoding as an array the request body will be a stdClass.

Also, you dont need to decode the body yourself, the WP_REST_Request class has a method to get the json parameters from the request. Look at the other api endpoint for an example: https://github.com/krokedil/ledyer-payments-for-woocommerce/blob/53365ec713570250e90f2e67f2e53310c28ea9ce/src/Callback.php#L73-L83